caffe2/perfkernels/embedding_lookup.cc

```diff
@@ -82,92 +82,92 @@ static void EmbeddingLookupGenericSlow(
-      const float* scale_bias,                                                             \
-      bool normalize_by_lengths,                                                           \
-      OutType* out) {                                                                      \
-    AVX2_FMA_DO(                                                                           \
-        EmbeddingLookup_##IndexType##_##InType##_##OutType##_##IS_WEIGHT_POSITIONAL,       \
-        block_size,                                                                        \
-        output_size,                                                                       \
-        index_size,                                                                        \
-        data_size,                                                                         \
-        input,                                                                             \
-        indices,                                                                           \
-        lengths,                                                                           \
-        weights,                                                                           \
-        scale_bias,                                                                        \
-        normalize_by_lengths,                                                              \
-        out);                                                                              \
-    BASE_DO(                                                                               \
-        EmbeddingLookup_##IndexType##_##InType##_##OutType##_##IS_WEIGHT_POSITIONAL,       \
-        block_size,                                                                        \
-        output_size,                                                                       \
-        index_size,                                                                        \
-        data_size,                                                                         \
-        input,                                                                             \
-        indices,                                                                           \
-        lengths,                                                                           \
-        weights,                                                                           \
-        scale_bias,                                                                        \
-        normalize_by_lengths,                                                              \
-        out);                                                                              \
+#define EMBEDDING_SPECIALIZATION(                                                                      \
+    IndexTypeName, IndexType, InTypeName, InType, OutTypeName, OutType, IS_WEIGHT_POSITIONAL)          \
```

Comment:
Hm?

Reply:
InType was used in function names, but we can't use at::Half there, so I made InTypeName to use in the function name and kept InType for the actual type. We don't really need IndexTypeName or OutTypeName because those are never Half, but I thought it would be strange not to have them.

e.g. ```EmbeddingLookup_##IndexTypeName##_##InTypeName##_##OutTypeName##_##IS_WEIGHT_POSITIONAL##__base```

Alternatively, we could just `using at::Half` in this file, but I kind of want it match `hp_emblookup_codegen.py`